frontend/src/Components/Charts/SearchRate/SearchRate.js

```diff
@@ -52,6 +54,32 @@ function SearchRate(props) {
   const renderMetaTags = useMetaTags();
   const [renderTableModal, { openModal }] = useTableModal();
 
+  const [referenceElement, setReferenceElement] = useState(null);
+  const [tooltipText, setTooltipText] = useState('');
+  const [popperElement, setPopperElement] = useState(null);
+  const { styles, attributes } = usePopper(referenceElement, popperElement, {
+    placement: 'top',
+    modifiers: [
+      {
+        name: 'offset',
+        options: {
+          offset: [0, 10],
+        },
+      },
+    ],
+  });
+
+  const showTooltip = (e) => {
+    setReferenceElement(e.currentTarget);
+    setTooltipText(tooltipLanguage(e.target.parentElement.id));
```

Comment:
Reason we need `parentElement` is the graphing library breaks up each label by word into span tags. The parent elem's id has the reference to the to stop purpose the user hovers over to then update the tooltip text.